File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    t, x, y = bandage
    end_time = attacks[-1][0] + 1
    
    idx, cnt, hp = 0, 0, health
    for time in range(1, end_time):
        if time == attacks[idx][0]:
            cnt = 0
            hp -= attacks[idx][1]
            if hp <= 0:
                return -1

            idx += 1
            if idx >= len(attacks):
                return hp
        else:
            hp += x
            
            cnt += 1
            if cnt == t:
                hp += y
                cnt = 0 
                
        if hp > health:
            hp = health                
    return hp

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int t = bandage[0]; int x = bandage[1]; int y = bandage[2];
        int endTime = attacks[attacks.length - 1][0] + 1;
        
        int hp = health; int idx = 0; int cnt = 0;
        for (int time = 1; time < endTime; time++) {
            if (time == attacks[idx][0]) {
                cnt = 0;
                
                hp -= attacks[idx++][1];
                if (hp <= 0) return -1;
                
                if (idx >= attacks.length) return hp;
            } else {
                hp += x;
                
                cnt += 1;
                if (cnt == t) {
                    hp += y;
                    cnt = 0;
                }
            }
            
            if (hp > health) hp = health;
        }
        return hp;
    }
}